swarm/network/stream/peer.go

```diff
@@ -420,3 +423,156 @@ func (p *Peer) close() {
 		s.Close()
 	}
 }
+
+// runUpdateSyncing is a long running function that creates the initial
+// syncing subscriptions to the peer and waits for neighbourhood depth change
+// to create new ones or quit existing ones based on the new neighbourhood depth
+// and if peer enters or leaves nearest neighbourhood by using
+// syncSubscriptionsDiff and updateSyncSubscriptions functions.
+func (p *Peer) runUpdateSyncing() {
```

Comment:
this function should be called from `Registry.Run`.
There `BzzAddr` is available from `BzzPeer` 
the remote overlay address should simply be passed as an argument to this function

Reply:
BzzPeer is encapsulated by stream Peer, so that BzzAdd is available, in the latest change.